pkg/cmd/server/origin/master.go

```diff
@@ -692,3 +693,34 @@ type clientDeploymentInterface struct {
 func (c clientDeploymentInterface) GetDeployment(ctx api.Context, name string) (*api.ReplicationController, error) {
 	return c.KubeClient.ReplicationControllers(api.NamespaceValue(ctx)).Get(name)
 }
+
+// namespacingFilter adds a filter that adds the namespace of the request to the context.  Not all requests will have namespaces,
+// but any that do will have the appropriate value added.
+func namespacingFilter(handler http.Handler, contextMapper kapi.RequestContextMapper) http.Handler {
+	infoResolver := &apiserver.APIRequestInfoResolver{util.NewStringSet("api", "osapi"), latest.RESTMapper}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		ctx, ok := contextMapper.Get(req)
+		if !ok {
+			http.Error(w, "Unable to find request context", http.StatusInternalServerError)
+			return
+		}
+
+		if _, exists := kapi.NamespaceFrom(ctx); !exists {
+			requestInfo, _ := infoResolver.GetAPIRequestInfo(req)
+
+			// TODO reconsider special casing this.  Having the special case hereallow us to fully share the kube
+			// APIRequestInfoResolver without any modification or customization.
+			namespace := requestInfo.Namespace
+			if (requestInfo.Resource == "projects") && (len(requestInfo.Name) > 0) {
+				namespace = requestInfo.Name
```

Comment:
Errors aren't caught on other weird conditions with non-namespaced resources having a namespace, so I'm going to with no.